src/esp_loader.ts

```diff
@@ -745,7 +747,7 @@ export class ESPLoader extends EventTarget {
     let timeout = 0;
     let buffer;
 
-    if (this.IS_STUB) {
+    if (!this.IS_STUB) {
       writeSize = size; // stub expects number of bytes here, manages erasing internally
```

Comment:
The if-check also checks for write size, not just timeout. You're changing how that is determined based on it being a stub now too.

Reply:
Yes you are correct - my mistake. I have now moved the timeout logic to be the other way around so the writeSize calculation is as it was previously.